indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentNukeAction.java

```diff
@@ -54,26 +56,33 @@ public Set<DataSegment> getSegments()
   }
 
   @Override
-  public TypeReference<Void> getReturnTypeReference()
+  public TypeReference<Set<DataSegment>> getReturnTypeReference()
   {
-    return new TypeReference<Void>()
+    return new TypeReference<Set<DataSegment>>()
     {
     };
   }
 
   @Override
-  public Void perform(Task task, TaskActionToolbox toolbox)
+  public Set<DataSegment> perform(Task task, TaskActionToolbox toolbox)
```

Comment:
Also, we are changing the semantics of an existing task action? Will this be a breaking change?

Reply:
The task action has been reverted to its original semantics.